### ISSUE TITLE
Duplicate log lines on CeleryExecutor not fixed with AIRFLOW__COSMOS__PROPAGATE_LOGS

### ISSUE THREAD

**Issue by adammarples** [astronomer/astronomer-cosmos#747]
I am running a celery executor on astronomer hosted, astronomer-cosmos==1.2.5, and getting duplicate log lines from astronomer-cosmos.

Eg.
```
[2023-12-05, 11:13:02 UTC] {log.py:232} WARNING - [[34m2023-12-05T11:13:02.130+0000[0m] {[34msubprocess.py:[0m73} INFO[0m - [35m(astronomer-cosmos)[0m - Running command: ['/usr/local/bin/dbt', '--warn-error', '--no-use-colors', 'run', '--models', 'MYTABLE', '--profiles-dir', '/tmp/tmppzx94vki', '--profile', 'myprofile', '--target', 'pr'][0m
[2023-12-05, 11:13:02 UTC] {subprocess.py:73} INFO - Running command: ['/usr/local/bin/dbt', '--warn-error', '--no-use-colors', 'run', '--models', 'MYTABLE', '--profiles-dir', '/tmp/tmppzx94vki', '--profile', 'myprofile', '--target', 'pr']
[2023-12-05, 11:14:10 UTC] {subprocess.py:94} INFO - 11:14:10  1 of 1 START hook: mypackage.on-run-start.0 ............................... [RUN]
[2023-12-05, 11:14:10 UTC] {subprocess.py:94} INFO - 11:14:10  1 of 1 OK hook: mypackage.on-run-start.0 .................................. [OK in 0.00s]
[2023-12-05, 11:14:10 UTC] {log.py:232} WARNING - [[34m2023-12-05T11:14:10.837+0000[0m] {[34msubprocess.py:[0m94} INFO[0m - [35m(astronomer-cosmos)[0m - 11:14:10  1 of 1 OK hook: mypackage.on-run-start.0 .................................. [OK in 0.00s][0m
[2023-12-05, 11:14:10 UTC] {log.py:232} WARNING - [[34m2023-12-05T11:14:10.838+0000[0m] {[34msubprocess.py:[0m94} INFO[0m - [35m(astronomer-cosmos)[0m - 11:14:10[0m
[2023-12-05, 11:14:10 UTC] {subprocess.py:94} INFO - 11:14:10
```

I have tried setting `AIRFLOW__COSMOS__PROPAGATE_LOGS = "False"` as per [this](https://github.com/astronomer/astronomer-cosmos/blob/e1f34ea424b8940f1c06660554298008876ea11b/docs/configuration/logging.rst#L8) and I have tried `AIRFLOW__COSMOS__PROPAGATE_LOGS=false` but neither have any effect on removing the astronomer-cosmos lines.

![image](https://github.com/astronomer/astronomer-cosmos/assets/22686431/b8049a6f-8db6-44bd-9497-af7faf10d6ab)

All the cosmos logs are WARNING level, which also seems strange.

**Comment by tatiana** [comment]
@adammarples, thanks for reporting the issue!

This is a bug, and the robot suggested using the configuration file to a workaround. Could you confirm this works?

The fix can be done along these lines:
https://github.com/astronomer/astronomer-cosmos/blob/6ccde5cf82fe61f0200b69eaa827cf4a0740bb16/tests/test_log.py#L20-L22

And we need add a test to make sure things work with env vars:
https://github.com/astronomer/astronomer-cosmos/blob/6ccde5cf82fe61f0200b69eaa827cf4a0740bb16/tests/test_log.py#L19-L24

Would you like to make this contribution?

**Comment by adammarples** [comment]
Hi @tatiana, I am just investigating it. It's my bad, it's not a bug, these log propagation changes work fine in main branch but are not included in version 1.2.5, so I'll wait for the update.

**Comment by tatiana** [comment]
@adammarples Sorry - we are really lacking versioned docs, I logged #750 so we don't lose track of this issue.

**Comment by tatiana** [comment]
@adammarples this change is available in the release 1.3.0a2, in case you'd like to try it out:
https://pypi.org/project/astronomer-cosmos/1.3.0a2/

Since it is a new feature, we didn't release as part of a 1.2 micro release. For some reason we missed to include it in our release notes, I'll fix this:
https://github.com/astronomer/astronomer-cosmos/releases/tag/astronomer-cosmos-v1.3.0a2
